### PR TITLE
OpenGL: Pool small textures, reuse handles

### PR DESCRIPTION
**Pull request by hrydgard**
Will hopefully work as a cleaner replacement for #16399  (Gran Turismo crashfix for Quest 2).

I guess we could periodically shrink the pool too, but since all textures in it are small, not sure it's worth the trouble.

This also deletes the unused nameCache_.

### CODE REVIEW

**Review by unknownbrackets — APPROVED** [review]

**Comment by hrydgard** [comment]
We don't actually build the GL files in UWP mode (though maybe we should, somehow) so that's actually not an issue here.

**Comment by unknownbrackets** [comment]
Ah, okay, just worried about files being added to everything but UWP.  But yeah, that makes sense.

-[Unknown]

**Comment by lvonasek** [comment]
I tested this PR and Gran Turismo still crashes in the menu (it is quite rare, a long test session needed to confirm it):
```
2022-12-04 17:02:06.962  5096-5096  DEBUG                   pid-5096                             A  *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
2022-12-04 17:02:06.962  5096-5096  DEBUG                   pid-5096                             A  Build fingerprint: 'oculus/hollywood/hollywood:10/QQ3A.200805.001/47421700959100000:user/release-keys'
2022-12-04 17:02:06.962  5096-5096  DEBUG                   pid-5096                             A  Revision: '0'
2022-12-04 17:02:06.962  5096-5096  DEBUG                   pid-5096                             A  ABI: 'arm64'
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A  Timestamp: 2022-12-04 17:02:06+0100
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A  pid: 4870, tid: 4940, name: AndroidRender  >>> org.ppsspp.ppsspp <<<
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A  uid: 10357
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A  signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x77ff000000
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A      x0  00000076cd2d0980  x1  00000076cd9a1280  x2  0000000000000034  x3  0000000000000100
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A      x4  0000000000000000  x5  00000076cd3e4220  x6  00000076cd9d5cc8  x7  00000076cd9d5cc8
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A      x8  0000000000000001  x9  0000000000000001  x10 00000076eb094980  x11 00000077ff000000
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A      x12 000000773d3dfe60  x13 00000076cd9d8138  x14 0000000000001b90  x15 00000000fdffffff
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A      x16 0000000000000000  x17 0000000000000000  x18 00000076d177e000  x19 00000076bb6d7c00
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A      x20 00000076cd9a1280  x21 00000076cd9d4080  x22 0000000000000001  x23 00000076bb6d7c00
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A      x24 0000000000000001  x25 00000076bb8a1880  x26 0000000000000000  x27 00000076bb6d7c00
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A      x28 0000000000000001  x29 00000076d19d4b90
2022-12-04 17:02:06.964  5096-5096  DEBUG                   pid-5096                             A      sp  00000076d19d4b70  lr  000000773d207da4  pc  000000773d232934
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A  backtrace:
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A        #00 pc 00000000002b0934  /vendor/lib64/egl/libGLESv2_adreno.so (!!!0000!bb42a6688242c365a0a7febd2e17e4!f9286358ab9!+140) (BuildId: d276693e032a36a839eb547ca0b029f9)
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A        #01 pc 0000000000285da0  /vendor/lib64/egl/libGLESv2_adreno.so (!!!0000!28f24188499da127c115ae7e78c710!f9286358ab9!+536) (BuildId: d276693e032a36a839eb547ca0b029f9)
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A        #02 pc 00000000002111a4  /vendor/lib64/egl/libGLESv2_adreno.so (!!!0000!58cbc5b86fa13da0d4e51c7edcc648!f9286358ab9!+548) (BuildId: d276693e032a36a839eb547ca0b029f9)
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A        #03 pc 000000000071d2b4  /data/app/org.ppsspp.ppsspp-oyL_-Q9sZMt_CfCsVxdqRw==/lib/arm64/libppsspp_jni.so (GLDeleter::Perform(GLRenderManager*, bool)+540) (BuildId: e6a31892702e6a0271e4c2b250c7d5a46cbf2578)
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A        #04 pc 000000000071e2b0  /data/app/org.ppsspp.ppsspp-oyL_-Q9sZMt_CfCsVxdqRw==/lib/arm64/libppsspp_jni.so (GLRenderManager::Run(GLRRenderThreadTask&)+108) (BuildId: e6a31892702e6a0271e4c2b250c7d5a46cbf2578)
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A        #05 pc 000000000071e140  /data/app/org.ppsspp.ppsspp-oyL_-Q9sZMt_CfCsVxdqRw==/lib/arm64/libppsspp_jni.so (GLRenderManager::ThreadFrame()+296) (BuildId: e6a31892702e6a0271e4c2b250c7d5a46cbf2578)
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A        #06 pc 000000000076a984  /data/app/org.ppsspp.ppsspp-oyL_-Q9sZMt_CfCsVxdqRw==/lib/arm64/libppsspp_jni.so (Java_org_ppsspp_ppsspp_NativeRenderer_displayRender+104) (BuildId: e6a31892702e6a0271e4c2b250c7d5a46cbf2578)
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A        #07 pc 000000000000908c  /data/app/org.ppsspp.ppsspp-oyL_-Q9sZMt_CfCsVxdqRw==/oat/arm64/base.odex (art_jni_trampoline+124)
2022-12-04 17:02:07.495  5096-5096  DEBUG                   pid-5096                             A        #08 pc 0000000002002030  /memfd:/jit-cache (deleted)
```

**Comment by unknownbrackets** [comment]
What if instead of glDeleteTextures for the large ones, we put a callback on the deleter to set to a 1x1 texture and then put it in the pool?

-[Unknown]

**Comment by hrydgard** [comment]
If it takes a long session of sitting around in GTs menus, at least it's still gonna be playable, but of course we should do what we can to fix it.

Yeah, that might be a sensible thing to try.

**Comment by hrydgard** [comment]
OK, now all textures will use the pool, big textures get shrunk when returned. Let's hope this makes the driver's life easy enough...

File: GPU/GLES/TextureCacheGLES.cpp [hunk 1]
```diff
@@ -297,6 +291,12 @@ void TextureCacheGLES::BuildTexture(TexCacheEntry *const entry) {
 		}
 	}
 
+	if (plan.depth == 1) {
+		entry->textureName = render_->CreateTexture(GL_TEXTURE_2D, tw, tw, 1, plan.levelsToCreate);
+	} else {
+		entry->textureName = render_->CreateTexture(GL_TEXTURE_3D, tw, tw, plan.depth, 1);
```
Comment (unknownbrackets) [inline GPU/GLES/TextureCacheGLES.cpp:297]: This isn't new, but is `tw, tw` actually correct?  not th?

-[Unknown]

**Review by unknownbrackets — APPROVED** [review]

**Comment by unknownbrackets** [comment]
Hm.  The first one is early in the destructor - use-after-free or double-free somehow?  But then the latter is JitBlockCache... just makes me worry about heap corruption again...

-[Unknown]

**Comment by hrydgard** [comment]
Ugh, spooky stuff. I also suspect memory corruption with that. Can't be sure whether it's the user-mode part of the driver or us that's causing the corruption though...

**Comment by hrydgard** [comment]
Considering closing this, as I've found memory errors and hoping that this fix is the solution instead: #16533

**Comment by lvonasek** [comment]
I confirm that the crashing does not happen in the master anymore.

**Comment by hrydgard** [comment]
Alright, then I'm just gonna abandon this, no need for it.

**Pull request closed without merging**